include/deal.II/matrix_free/tensor_product_kernels.h

```diff
@@ -3122,7 +3122,7 @@ namespace internal
           {
             const Number2 test_value_y1_2 = value[1] * p[1];
             const Number2 test_value_y0_2 = value[1] - test_value_y1_2;
-            const Number2 value0_2        = p[0] * test_value_y1_2;
```

Comment:
Why is this not tested?

Reply:
https://github.com/dealii/dealii/blob/e94b7878ce4e89f6e1015828e0bfa21fd74ad11d/include/deal.II/matrix_free/fe_point_evaluation.h#L1798-L1801

That's why. I am currently rethinking this condition. That's also how I discovered the bug. I can add a test with a high number of quadrature points which should trigger this.

Reply:
> I can add a test with a high number of quadrature points which should trigger this.

That is a good idea :+1:

Reply:
The number of points was already high enough, we simply did not check the linear 3D path. The adjusted test fails on master and passes on this branch.

Reply:
Excellent. Thank you!